app/controllers/vacancies_controller.rb

```diff
@@ -24,7 +24,14 @@ def index
   end
 
   def show
-    vacancy = Vacancy.listed.friendly.find(id)
+    begin
+      vacancy = Vacancy.listed.friendly.find(id)
+    rescue ActiveRecord::RecordNotFound
+      raise unless Vacancy.trashed.friendly.exists?(id)
+
+      return render '/errors/trashed_vacancy_found'
```

Comment:
Also, can you add `status: :not_found` here too, so the server returns the correct status code?

Reply:
:+1: